Sources/NIOHTTP2/HTTP2PipelineHelpers.swift

```diff
@@ -460,40 +556,58 @@ extension Channel {
     /// Configures a `ChannelPipeline` to speak either HTTP or HTTP/2 according to what can be negotiated with the client.
     ///
     /// This helper takes care of configuring the server pipeline such that it negotiates whether to
-    /// use HTTP/1.1 or HTTP/2. Once the protocol to use for the channel has been negotiated, the
-    /// provided callback will configure the application-specific handlers in a protocol-agnostic way.
+    /// use HTTP/1.1 or HTTP/2.
     ///
     /// This function doesn't configure the TLS handler. Callers of this function need to add a TLS
     /// handler appropriately configured to perform protocol negotiation.
     ///
     /// - Parameters:
-    ///   - connectionConfiguration: The settings that will be used when establishing the connection. These will be sent to the peer as part of the
-    ///         handshake.
-    ///   - streamConfiguration: The settings that will be used when establishing new streams. These mainly pertain to flow control.
+    ///   - connectionConfiguration: The settings that will be used when establishing the HTTP/2 connection.
+    ///         These will be sent to the peer as part of the handshake.
+    ///   - streamConfiguration: The settings that will be used when establishing new HTTP/2 streams.
+    ///         These mainly pertain to flow control.
     ///   - streamDelegate: The delegate to be notified in the event of stream creation and close.
+    ///   - h1ConnectionChannelConfigurator: An optional callback that will be invoked only
+    ///         when the negotiated protocol is HTTP/1.1 to configure the connection channel.
     ///   - h2ConnectionChannelConfigurator: An optional callback that will be invoked only
-    ///         when the negotiated protocol is H2 to configure the connection channel.
+    ///         when the negotiated protocol is HTTP/2 to configure the connection channel.
     ///   - inboundStreamInitializer: A closure that will be called whenever the remote peer initiates a new stream.
```

Comment:
This doc comment doesn't match with the parameter names

Reply:
I _think_ they do but that the diff is presenting them in a confusing way

Reply:
I restructured the original commit to split out the simple moving of a function to group like functions. Hopefully that makes the main diff easier to understand